### ISSUE TITLE
'cannot be instantiated, class is defined as private' false positive when friend is a superclass

### ISSUE THREAD

**Issue by fdupret** [abaplint/abaplint#3884]
```abap
CLASS lcl_super_factory DEFINITION.
  PUBLIC SECTION.
ENDCLASS.

CLASS lcl_super_factory IMPLEMENTATION.

ENDCLASS.

CLASS lcl_class DEFINITION
  CREATE PRIVATE
  FRIENDS lcl_super_factory.

  PUBLIC SECTION.
  PROTECTED SECTION.
  PRIVATE SECTION.

ENDCLASS.

CLASS lcl_class IMPLEMENTATION.


ENDCLASS.

CLASS lcl_sub_factory DEFINITION
  INHERITING FROM lcl_super_factory
  CREATE PUBLIC.

  PUBLIC SECTION.
    METHODS get_instance
      RETURNING
        VALUE(result) TYPE REF TO lcl_class.
  PROTECTED SECTION.
  PRIVATE SECTION.

ENDCLASS.

CLASS lcl_sub_factory IMPLEMENTATION.

  METHOD get_instance.
    CREATE OBJECT result TYPE lcl_class.
  ENDMETHOD.

ENDCLASS.
```

This is allowed in ABAP but abaplint returns `lcl_class cannot be instantiated, class is defined as private(check_syntax)`

**Comment by larshp** [comment]
@fdupret try again, 2.118.9 published

**Comment by fdupret** [comment]
Fixed, thanks!
